networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1139 @@
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+
+    See Also
+    --------
+    treeness, feedforwardness, hierarchy_coordinates
+    """
+    import numpy as np
+
+    if not G.is_directed():
+        raise nx.NetworkXError(
+            "G must be a directed graph for any hierarchy coordinate evaluation"
+        )
+    # unweighted (non-binary) check
+    if not np.array_equal(np.unique(nx.to_numpy_array(G)), [0, 1]):
+        condensed_graphs, original_graphs = node_weighted_condense(
+            nx.to_numpy_array(G),
+            num_thresholds,
+            threshold_distribution,
+        )
+        o = sum(
+            orderability(OG, CG) for CG, OG in zip(condensed_graphs, original_graphs)
+        )
+        return o / len(condensed_graphs)
+
+    if condensed_nx_graph is None:
+        condensed_nx_graph = weight_nodes_by_condensation(nx.condensation(G))
+    weights = nx.get_node_attributes(condensed_nx_graph, "weight").values()
+    total_acyclic_node_weight = sum(weights)
+    number_non_cyclic_nodes = sum(1 for wt in weights if wt == 1)
+    return number_non_cyclic_nodes / total_acyclic_node_weight
+
+
+@not_implemented_for("undirected")
+def _feedforwardness_iteration(G):
+    """Performs a single iteration of the feedforward algorithm described in [1]_
+
+    Parameters
+    ----------
+    G: NetworkX Graph
+        a directed, acyclic graph [1]_
+
+    Return
+    --------
+    g: float
+        sum of feedforwardness for a single Directed Acyclic Graph
+    num_paths: int
+        sum total paths considered
+
+    Notes
+    ______
+    g(G) = sum of F over all paths from maximal to minimal nodes
+
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+
+    See Also
+    --------
+    feedforwardness
+    """
+    if not nx.is_directed_acyclic_graph(G):
+        raise nx.NetworkXError(
+            "G must be a directed acyclic graph for feedforwardness evaluation"
+        )
+
+    max_layer = max_min_layers(G, max_layer=True)
+    min_layer = max_min_layers(G, max_layer=False)
+    weights = nx.get_node_attributes(G, "weight")
+    g = 0
+    num_paths = 0
+    for max_node in max_layer:
+        for min_node in min_layer:
+            for path in nx.all_simple_paths(G, source=max_node, target=min_node):
+                # where each path calculation is F(path)
+                g += len(path) / sum([weights[node] for node in path])
+                num_paths += 1
+    return g, num_paths
+
+
+@not_implemented_for("undirected")
+def feedforwardness(DAG):
+    """Returns feedforwardness hierarchy coordinates, ~extent information passes vertically
+    (vs horizontally) within a hierarchy.
```

Comment:
The one-line description is used to labels documentation links to this function.
So it really should be short and sweet (one line:)

Here, you should split this an start a next paragraph with the description of what feedforwardness is.
Then  the paragrph you've got about how it is computed -- which should be clarified... it is too vague
as currently written.  What are the weights, how is the "repeated calculation" combined to get a single value (or is it combined?)

Reply:
I wrote up something more of a description, though perhaps the added math is a bit much. Let me know if the description could be improved.